src/julia/modules/Zygote/ZygoteLSTM.jl

```diff
@@ -0,0 +1,111 @@
+module ZygoteLSTM
+include("../../shared/load.jl")
+using ADPerfTest
+using LSTMData
+using Zygote
+
+export get_lstm_test
+
+# objective
+function sigmoid(x::Float64)
+    1. / (1. + exp(-x))
+end
+
+function lstmmodel(weight::T1, bias::T2, hidden::T3, cell::T4, input::T5)::Tuple{Vector{Float64}, Vector{Float64}} where {T1<:AbstractVector{Float64}, T2<:AbstractVector{Float64}, T3<:AbstractVector{Float64}, T4<:AbstractVector{Float64}, T5<:AbstractVector{Float64}}
+    hsize = size(hidden, 1)
+    forget = sigmoid.(input .* view(weight, 1:hsize) .+ view(bias, 1:hsize))
+    ingate = sigmoid.(hidden .* view(weight, hsize+1:2hsize) .+ view(bias, hsize+1:2hsize))
+    outgate = sigmoid.(input .* view(weight, 2hsize+1:3hsize) .+ view(bias, 2hsize+1:3hsize))
+    change = tanh.(hidden .* view(weight, 3hsize+1:4hsize) .+ view(bias, 3hsize+1:4hsize))
+
+    cell2 = cell .* forget .+ ingate .* change
+    hidden2 = outgate .* tanh.(cell2)
+    hidden2, cell2
+end
+
+function lstmpredict(main_params::Matrix{Float64}, extra_params::Matrix{Float64}, state::Matrix{Float64}, input::T)::Tuple{Vector{Float64}, Matrix{Float64}} where {T<:AbstractVector{Float64}}
+    x1 = input .* view(extra_params, :, 1)
+    x = view(x1, :)
+    lenstate = size(state, 2)
+    s2 = Matrix{Float64}(undef, size(input, 1), 0)
+    for i ∈ 1:2:lenstate
+        h, c = lstmmodel(view(main_params, :, i), view(main_params, :, i + 1), view(state, :, i), view(state, :, i + 1), x)
+        x = h
+        # Zygote does not support mutating arrays
+        # and I didn't manage to rewrite this into an array comprehension
```

Comment:
Perhaps add TODO: rewrite as array comprehension - initial attempts were not successfully differentiated by Zygote

Reply:
Done.